iml-agent/src/action_plugins/ltuer.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use std::path::{Path, PathBuf};
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+use tokio_fs as fs;
+
+static CONFIGURATION_DIR: &str = "/etc/iml";
+
+async fn create_ltuer_conf_internal<W>(
+    (mailbox_path, fs_name, cold_pool): (String, String, String),
+    writer: &mut W,
+) -> Result<(), ImlAgentError>
+where
+    W: AsyncWrite + Unpin,
+{
+    let contents = format!(
+        "mailbox={}\nfs_name={}\ncold_pool={}\n",
+        mailbox_path, fs_name, cold_pool
+    );
+
+    writer
+        .write(contents.as_bytes())
+        .await
+        .map(|_| ())
+        .map_err(|e| e.into())
+}
+
+async fn wrap_file_creation<P>(
+    (mailbox_path, fs_name, cold_pool): (String, String, String),
+    dir_path: P,
+) -> Result<(), ImlAgentError>
```

Comment:
For me the name is suboptimal. Essentially it is `create_ltuer_conf_internal`, and the function above, which is writing stuff is something like `write_ltuer_conf`.

Reply:
Good suggestion, renamed